### PR TITLE
AVFoundationPlayback code climate issues fix

### PR DESCRIPTION
**Pull request by caiobzen**
## 🏁  Goal: 
To solve some of the minor Code Climate issues we have in this file.
We know that AVFoundationPlayback needs a deeper refactoring strategy, so this pull request is aiming to help us out in the future and not to solve the root of the problem.

## ✅  How to test:
- Since this is just some refactoring, run the tests and nothing should break.
- You can even try out some backward/forward seek since there were some changes around this code.

Captura de Tela 2019-10-11 às 16 54 20

### CODE REVIEW

File: Sources/Clappr/Classes/Plugin/Playback/AVFoundationPlayback.swift [hunk 1]
```diff
@@ -203,19 +203,15 @@ open class AVFoundationPlayback: Playback {
 
     open override func play() {
         guard canPlay else { return }
-
-        if player == nil {
-            setupPlayer()
-        }
-
+        setupPlayerIfNeeded()
```
Comment (paulogama14) [inline Sources/Clappr/Classes/Plugin/Playback/AVFoundationPlayback.swift]: It's a good point in a better way, but I think we are adding more than one responsibility to this method.

File: Sources/Clappr/Classes/Plugin/Playback/AVFoundationPlayback.swift [hunk 2]
```diff
@@ -203,19 +203,15 @@ open class AVFoundationPlayback: Playback {
...
         trigger(.willPlay)
         player?.play()
+        updateInitialStateIfNeeded()
```
Comment (paulogama14) [inline Sources/Clappr/Classes/Plugin/Playback/AVFoundationPlayback.swift]: Some as above.

File: Sources/Clappr/Classes/Extension/AVPlayerItem+Ext.swift [hunk 3]
```diff
@@ -4,4 +4,14 @@ extension AVPlayerItem {
     func isFinished(with threshold: TimeInterval = 2.0) -> Bool {
         return fabs(CMTimeGetSeconds(duration) - CMTimeGetSeconds(currentTime())) <= threshold
     }
+
+    func seek(to timeInterval: TimeInterval, _ completion: (() -> Void)? = nil) {
```
Comment (paulogama14) [inline Sources/Clappr/Classes/Extension/AVPlayerItem+Ext.swift]: Is it valid to test this separately?

File: Sources/Clappr/Classes/Plugin/Playback/AVFoundationPlayback.swift (hunk 1 above)
Comment (caiobzen) [reply Sources/Clappr/Classes/Plugin/Playback/AVFoundationPlayback.swift]: the more than one responsibility was already there. The goal of this pull request is to reduce the `cognitive complexity` around it.

File: Sources/Clappr/Classes/Plugin/Playback/AVFoundationPlayback.swift (hunk 2 above)
Comment (caiobzen) [reply Sources/Clappr/Classes/Plugin/Playback/AVFoundationPlayback.swift]: Same as above

**Pull request merged**